src/control/server/ctl_storage_rpc.go

```diff
@@ -61,73 +70,118 @@ func newResponseState(inErr error, badStatus ctlpb.ResponseStatus, infoMsg strin
 	return rs
 }
 
-// stripNvmeDetails removes all controller details leaving only PCI address and
-// NUMA node/socket ID. Useful when scanning only device topology.
-func stripNvmeDetails(pbc *ctlpb.NvmeController) {
-	pbc.Serial = ""
-	pbc.Model = ""
-	pbc.FwRev = ""
+// Convert bdev scan results to protobuf response.
+func bdevScanToProtoResp(scan scanBdevsFn, req storage.BdevScanRequest) (*ctlpb.ScanNvmeResp, error) {
+	resp, err := scan(req)
+	if err != nil {
+		return nil, err
+	}
+
+	pbCtrlrs := make(proto.NvmeControllers, 0, len(resp.Controllers))
+
+	return &ctlpb.ScanNvmeResp{
+		State:  new(ctlpb.ResponseState),
+		Ctrlrs: pbCtrlrs,
+	}, pbCtrlrs.FromNative(resp.Controllers)
```

Comment:
Should we be returning the resp even if there's an error?

Reply:
I don't really see the problem with doing so given we check for error before using a response in all the callers.

Reply:
done in the next commit

Reply:
It is certainly best practice to check that the returned error was non-nil before using the response, but best practices aren't always followed. Better to return a nil response for errors so that any programming mistake is detected quickly during development (hopefully!).